core/controllers/classifier.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Controllers for communicating with the VM for training classifiers."""
+
+import hashlib
+import hmac
+import json
+
+from core.controllers import base
+from core.domain import classifier_domain
+from core.domain import classifier_services
+from core.domain import config_domain
+
+import feconf
+
+
+def validate_request(handler):
+    """Decorator that checks if the incoming request for storing trained
+    classifier is valid."""
+    def test_is_valid(self, **kwargs):
+        message = self.request.get('payload')
+        message = json.loads(message)
+        signature = message.get('signature')
+        vm_id = message.get('vm_id')
+        message.pop('signature')
+        message = json.dumps(sorted(message))
```

Comment:
You should use `json.dumps({key: message[key] for key in sorted(message)})`. But even after doing this I am not sure it would maintain the order. Basically dicts are desgined to be order-less because they use hashing function in the back. So we ought to use orderdDict here, I guess.